internal/super/generate.go

```diff
@@ -122,64 +136,60 @@ func generateScript(
 	_ flowkit.Services,
 	state *flowkit.State,
 ) (result command.Result, err error) {
-	return generateNew(args, "script", generateFlags.Directory, logger, state)
-}
-
-func addCDCExtension(name string) string {
-	if strings.HasSuffix(name, ".cdc") {
-		return name
+	options := GeneratorOptions{
+		Directory: "cadence",
+		State:     state,
+		Logger:    logger,
 	}
-	return fmt.Sprintf("%s.cdc", name)
+	generator := NewGenerator(options)
+	err = generator.Create(TemplateMap{"script": args[0]})
+	return nil, err
 }
 
-func stripCDCExtension(name string) string {
-	return strings.TrimSuffix(name, filepath.Ext(name))
+// TemplateMap defines a map of template types to their specific names
+type TemplateMap map[string]string
+
+type GeneratorOptions struct {
+	Directory   string
+	State       *flowkit.State
+	Logger      output.Logger
+	DisableLogs bool
 }
```

Comment:
Imo this is unnecessary & belongs in the `Generator` struct itself, and these can be set in the factory method.  Defining options like this isn't necessarily any safer/different than:

```
g := &Generator{
	Directory: xxx,
	State: xxx,
	Logger: xxx,
	DisableLogs: xxx,
}
```

And `GeneratorOptions` probably just adds an unnecessary layer of complexity.  The factory method can be used to enforce that a `&Generator` object is created with valid configuration.

e.g. `NewGenerator(directory, state, logger)`, `NewGeneratorWithoutLogs(directory, state)` -- although I _think_ that NOOP loggers is the idiomatic way to handle this.